src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -190,37 +190,46 @@ case class WorkflowActor(workflow: WorkflowDescriptor,
   */
   onTransition {
     case fromState -> toState =>
-      log.info(s"$tag transitioning from $fromState to $toState.")
-      globalDataAccess.updateWorkflowState(workflow.id, toState)
-      /*
-        Send a message to self to trigger an actor shutdown. Run on a short timer to help enable some
-        unit test instrumentation
-       */
-      if (toState.isTerminal) {
-        backend.cleanUpForWorkflow(workflow)
-        globalDataAccess.updateWorkflowOptions(workflow.id, workflow.workflowOptions.clearEncryptedValues)
-        setTimer(s"WorkflowActor termination message: $tag", Terminate, AkkaTimeout, DontRepeatTimer)
+      def handleTerminalWorkflow: Future[Unit] = {
+        for {
+          _ <- backend.cleanUpForWorkflow(workflow)
+          _ <- globalDataAccess.updateWorkflowOptions(workflow.id, workflow.workflowOptions.clearEncryptedValues)
+          //  Send a message to self to trigger an actor shutdown. Run on a short timer to help enable some
+          //  unit test instrumentation
+          _ = setTimer(s"WorkflowActor termination message: $tag", Terminate, AkkaTimeout, DontRepeatTimer)
+        } yield ()
       }
+
+      for {
+        // Write the new workflow state before logging the change, tests assume the change is in effect when
+        // the message is logged.
+        _ <- globalDataAccess.updateWorkflowState(workflow.id, toState)
+        _ = log.info(s"$tag transitioning from $fromState to $toState.")
+        _ <- handleTerminalWorkflow if toState.isTerminal
+      } yield ()
   }
 
   private def persistStatus(key: ExecutionStoreKey, status: ExecutionStatus,
-                            returnCode: Option[Int] = None): Future[Unit] = {
-    persistStatuses(Iterable(key), status, returnCode)
+                            returnCode: Option[Int] = None): Unit = {
+    Await.result(persistStatuses(Iterable(key), status, returnCode), Duration.Inf)
```

Comment:
I'm ok w/ that provided the implication is that the ticket would be immediately (or at least measured in days, not weeks) picked up as opposed to the other "we shouldn't be blocking" ticket which likely won't see the light of day until Haley's Comet comes around again